File: app/src/main/java/com/maxwellscore/sparkone/data/network/WeatherService.java
package com.maxwellscore.sparkone.data.network;

import com.maxwellscore.sparkone.data.dto.WeatherData;

import io.reactivex.rxjava3.core.Observable;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;

public interface WeatherService {
    @GET("/weather")
    Observable<WeatherData> getWeather();

    @POST("/location")
    Observable<String> postLocation(@Body String location);
}

File: kotlinexample/src/main/java/com/maxwellscore/kotlinexample/data/WeatherService.kt
package com.maxwellscore.kotlinexample.data

import com.maxwellscore.kotlinexample.data.dto.WeatherData
import io.reactivex.rxjava3.core.Completable
import io.reactivex.rxjava3.core.Observable
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST

interface WeatherService {

    @GET("/weather")
    fun getWeather(): Observable<WeatherData>

    @POST("/location")
    fun postLocatiton(@Body location: String): Completable
}